memstore/purge.go

```diff
@@ -17,10 +17,18 @@ package memstore
 import (
 	memCom "github.com/uber/aresdb/memstore/common"
 	"github.com/uber/aresdb/utils"
+	"github.com/pkg/errors"
 )
 
 // Purge purges out of retention data for table shard
 func (m *memStoreImpl) Purge(tableName string, shardID, batchIDStart, batchIDEnd int, reporter PurgeJobDetailReporter) error {
+	// check if there is peer bootstraping job running, skip this purge if we can not acquire the token
+	if m.options.bootstrapToken.AcquireToken(tableName, uint32(shardID)) {
```

Comment:
do we need to AcquireToken for other processes as well (snapshot, archive) ?

Reply:
yes, both snapshot/archiving/backfill covered